getting_started/scripting/gdscript/gdscript_styleguide.rst

```diff
@@ -194,3 +207,44 @@ Constants
 
 Use CONSTANT\_CASE, all caps, with an underscore (\_) to separate words:
 ``const MAX_SPEED = 200``
+
+Static typing
+-------------
+
+Since Godot 3.1, GDScript supports :ref:`optional static typing<doc_gdscript_static_typing>`.
+
+Type hints
+~~~~~~~~~~
+
+Place the colon right after the variable's name, without a space, and let the GDScript compiler infer the variable's type when possible.
+
+
+**Good**:
+
+::
+
+   onready var health_bar: ProgressBar = get_node("UI/LifeBar")
+
+   var health := 0 # The compiler will use the int type
+
+**Bad**:
+
+::
+
+   # The compiler can't infer the exact type and will use Node 
+   # instead of ProgressBar
+   onready var health_bar: = get_node("UI/LifeBar") 
+
+   var health: int = 0
```

Comment:
I'm not sure I would show this one as "bad". It's ok to recommend using `:=` for primitive types to let the compiler infer the type, but I think using `: int` explicitly should still be valid in the style guide.

Reply:
Makes sense, just removed that.